src/components/Modal.jsx

```diff
@@ -52,6 +53,42 @@ const animations = {
     },
 };
 
+export const useLocation = () => {
```

Comment:
i don't think UI Kit should listen to window.location events, since this can be used across many apps and the Modal component should not control its behaviour 

This Component should only be responsible to show the content provide and close the modal incase parent passes onClose option

You need to move this logic lo x2-seller app.